### PR TITLE
Adding a sample showing a UoW leaning on child containers to store the session

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

**Comment by SimonCropp** [comment]
i dont think this should be called "structure map" we will not be taking the vector and producing samples for "UOW+every other DI". If people want that they can mix the UOW sample with the specific DI samples.

So propose to rename to something like "leveraging DI in a UOW"?

**Comment by andreasohlund** [comment]
Good point, I'll rename and try to wrap up the walkthrough today

**Comment by andreasohlund** [comment]
@Particular/documentation this one is ready for review

File: samples/unit-of-work/using-childcontainers/Version_5/Sample/Order.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿class Order
+{
+    public string Id { get; set; }
+
+    public string OrderNumber { get; set; }
+    public double OrderValue { get; set; }
+
+    public Order(string orderNumber, double orderValue)
```
Comment (SimonCropp) [inline samples/unit-of-work/using-childcontainers/Version_5/Sample/Order.cs]: why the constructor if u have settable props?

File: samples/unit-of-work/using-childcontainers/Version_5/Sample/Program.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using NServiceBus;
+using Raven.Client;
+using Raven.Client.Document;
+using StructureMap;
+
+static class Program
+{
+    static void Main()
+    {
+        IDocumentStore store = new DocumentStore()
```
Comment (SimonCropp) [inline samples/unit-of-work/using-childcontainers/Version_5/Sample/Program.cs]: dispose this

File: samples/unit-of-work/using-childcontainers/Version_5/Sample/Program.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+     
+        BusConfiguration configuration = new BusConfiguration();
+        configuration.EndpointName("Samples.StructureMap");
+
+        #region ContainerConfiguration
+        Container container = new Container(x =>
+        {
+            x.For<IDocumentStore>().Use(store);
+            x.For<IDocumentSession>().Use(c => c.GetInstance<IDocumentStore>().OpenSession());
```
Comment (SimonCropp) [inline samples/unit-of-work/using-childcontainers/Version_5/Sample/Program.cs]: why do you resolve IDocumentStore when u have the single instance? and why is store needed in the container? dont u only need the session?

File: samples/unit-of-work/using-childcontainers/Version_5/Sample/Program.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        IDocumentStore store = new DocumentStore()
+        {
+            Url = "http://localhost:32076",
+            DefaultDatabase = "Samples.StructureMapUoW"
+        }.Initialize();
+
+     
+        BusConfiguration configuration = new BusConfiguration();
+        configuration.EndpointName("Samples.StructureMap");
```
Comment (SimonCropp) [inline samples/unit-of-work/using-childcontainers/Version_5/Sample/Program.cs]: bad endpoint name.

File: samples/unit-of-work/using-childcontainers/sample.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+This sample shows how to share a RavenDB `IDocumentSession` between multiple handlers and finally call `.SaveChanges` once per message. This has the following benefits:
+
+1. Applies the changes in one round trip to the database
+2. Avoids the need to call `.SaveChanges` in all handlers
+
+With the UoW in place our handlers can focus solely on the business problem we're solving. Our handler in the sample looks like this:
+
+<!-- import PlaceOrderHandler -->
+
+Notice how we just inject the `IDocumentSession` and then just call `.Store` on it.
```
Comment (SimonCropp) [inline samples/unit-of-work/using-childcontainers/sample.md]: "just" is redundant

File: samples/unit-of-work/using-childcontainers/sample.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+The final piece of the puzzle is disposing of the session. Again the just works out of the box since the child container will automatically dispose any instances that was created by it.
+
+With this in mind we just configure our StructureMap container as follows
+
+<!-- import ContainerConfiguration -->
+
+Notice how we define the session factory as `single instance` and the session as `per call`.
+
+That's all we need!
+
```
Comment (SimonCropp) [inline samples/unit-of-work/using-childcontainers/sample.md]: why the white space?

File: samples/unit-of-work/using-childcontainers/sample.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+#### Registering the UoW in the pipeline
+
+This is a easy as
+
+<!-- import PipelineRegistration -->
+
+#### Sharing the session between handlers and behavior
+
+For this we're leaning on the child container support of our container. This sample is using StructureMap but it would work on most other containers as well. (please check their documentation to make sure)
```
Comment (SimonCropp) [inline samples/unit-of-work/using-childcontainers/sample.md]: > please check the documentation for your chosen container to make sure

File: samples/unit-of-work/using-childcontainers/sample.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+
+#### Saving the changes
+With RavenDB you need to call `.SaveChanges` explicitly. We'll add a pipeline step that does this once after all our handlers have been called.
+
+In this sample we'll use a incoming pipeline behavior to handler this.
+
+<!-- import RavenUnitOfWork -->
+
+Notice how we first call `next()` to invoke the pipeline and if no exception is thrown we'll just call `.SaveChanges()` to apply the changes to our database.
```
Comment (SimonCropp) [inline samples/unit-of-work/using-childcontainers/sample.md]: `just` is redundant

**Comment by andreasohlund** [comment]
Thanks for the review @SimonCropp . Pull has been updated

**Pull request merged**